### PR TITLE
core: add `bridge_count_connected_interfaces` to check if bridge has any connected interface

### PR DESCRIPTION
**Pull request by flouthoc**
add helper to check if bridge has any connected interface

Usage
```rust
    //count interfaces on bridge
    if let Ok(vector) =  bridge_count_connected_interfaces("cni-podman0") {
        println!("{:?}", vector.len());
    }

```

### CODE REVIEW

**Review by flouthoc — COMMENTED** [review]
PR has poor error handling, please wait for me to correct error.

**Comment by flouthoc** [comment]
@Luap99 @mheon @baude This is good to merge. I handled the error now.

File: src/network/core_utils.rs [hunk 1]
```diff
@@ -34,6 +34,99 @@ impl CoreUtils {
         final_slice.join(":")
     }
 
+    //count interfaces on bridge
+    #[tokio::main]
+    #[allow(irrefutable_let_patterns)]
```
Comment (Luap99) [inline src/network/core_utils.rs:39]: Why is this needed?

File: src/network/core_utils.rs [hunk 2]
```diff
@@ -34,6 +34,99 @@ impl CoreUtils {
...
+    pub async fn bridge_count_connected_interfaces(
+        ifname: &str,
+    ) -> Result<Vec<u32>, std::io::Error> {
```
Comment (Luap99) [inline src/network/core_utils.rs:42]: Why do we return an vector, it should be enough to return a bool.

File: src/network/core_utils.rs [hunk 3]
```diff
@@ -34,6 +34,99 @@ impl CoreUtils {
...
+        while let msg = links.try_next().await {
+            match msg {
+                Ok(Some(msg)) => {
+                    for nla in msg.nlas.into_iter() {
+                        if let Nla::Master(data) = nla {
+                            if data == master_id {
+                                if msg.header.index == master_index {
+                                    continue;
+                                }
+                                connected_veth.push(msg.header.index);
```
Comment (Luap99) [inline src/network/core_utils.rs:106]: just return true here if we have a match, see other comment

File: src/network/core_utils.rs [hunk 4]
```diff
@@ -34,6 +34,99 @@ impl CoreUtils {
...
+        tokio::spawn(_connection);
+
+        let mut master_id: u32 = 0;
+        let mut master_index: u32 = 0;
+
+        let mut links = handle
+            .link()
+            .get()
+            .set_name_filter(ifname.to_string())
+            .set_filter_mask(AF_BRIDGE as u8, RTEXT_FILTER_BRVLAN)
```
Comment (Luap99) [inline src/network/core_utils.rs:63]: Why do you need this filter here?

File: src/network/core_utils.rs [hunk 5]
```diff
@@ -34,6 +34,99 @@ impl CoreUtils {
...
+            .execute();
+        while let msg = links.try_next().await {
```
Comment (Luap99) [inline src/network/core_utils.rs:65]: Why is this a loop, in the other code parts you use to get a single interface
```
let mut links = handle
            .link()
            .get()
            .set_name_filter(macvlan_tmp_ifname.to_string())
            .execute();
        match links.try_next().await {
...
```

File: src/network/core_utils.rs (hunk 2 above)
Comment (flouthoc) [reply src/network/core_utils.rs:42]: I just thought in future if need to query connected `veth`'s  then function should support so. So it returns a list of  id's of connected veth's

File: src/network/core_utils.rs (hunk 4 above)
Comment (flouthoc) [reply src/network/core_utils.rs:63]: This is not needed on the first block i can remove this.

File: src/network/core_utils.rs (hunk 5 above)
Comment (flouthoc) [reply src/network/core_utils.rs:65]: Yes the first loop is not needed. I amend this.

File: src/network/core_utils.rs (hunk 3 above)
Comment (flouthoc) [reply src/network/core_utils.rs:106]: reasoning in the above comment.

File: src/network/core_utils.rs (hunk 2 above)
Comment (Luap99) [reply src/network/core_utils.rs:42]: I do not see why we need this and this will prevent us from an early exit when we found one interface.

Comment (flouthoc) [reply src/network/core_utils.rs:42]: no issues I'll convert this to `bool` if any interface is found.
